app/src/main/java/edu/wpi/first/shuffleboard/app/WidgetTileController.java

```diff
@@ -34,20 +28,31 @@
   @FXML
   private void initialize() {
     tile.addEventHandler(MouseEvent.MOUSE_CLICKED, this::changeView);
-    sourceProperty = EasyBind.monadic(tile.widgetProperty()).selectProperty(Widget::sourceProperty);
-    titleLabel.textProperty().bind(sourceProperty.selectProperty(DataSource::nameProperty));
+    tile.widgetProperty().addListener((__, oldWidget, widget) -> {
+      setTitleFor(widget);
+      widget.getSources().addListener((InvalidationListener) o -> setTitleFor(widget));
+      widget.getSources().addListener((InvalidationListener) o -> {
+        boolean noSources = widget.getSources().stream().allMatch(s -> s instanceof DestroyedSource);
+        tile.pseudoClassStateChanged(NO_SOURCE, noSources);
+        tile.setDisable(noSources);
+      });
+    });
+    setTitleFor(tile.getWidget());
     tile.centerProperty().bind(
         Bindings.createObjectBinding(
             this::createCenter, tile.widgetProperty(), tile.showWidgetProperty()));
-    sourceProperty.addListener((__, oldSource, newSource) -> {
-      if (newSource instanceof DestroyedSource) {
-        tile.pseudoClassStateChanged(NO_SOURCE, true);
-        tile.setDisable(true);
-      } else {
-        tile.pseudoClassStateChanged(NO_SOURCE, false);
-        tile.setDisable(false);
-      }
-    });
+  }
+
+  private void setTitleFor(Widget widget) {
```

Comment:
this setTileFor stuff needs to be re-worked for Widgets that control their own titles using a titleProperty